test/unit/puppetlabs/puppetserver/ruby/http_client_test.clj

```diff
@@ -126,17 +126,27 @@
              {:cipher-suites ~cipher-suites}))
     ~@body))
 
+(defn raise-caught-http-error
+  "Returns a Ruby script string that executes `script', rescues the expected
+   Puppet::Server::HttpClientError, and raises the nested exception for testing
+   the underlying cause."
+  [script]
+  (str "begin;"
+       script
+       ";rescue Puppet::Server::HttpClientError => e;"
+       "  raise e.cause.getCause;"
+       "end"))
+
 (deftest https-tls-defaults
   (testing "requests fail without an SSL client"
     (with-webserver-with-protocols nil nil
-       (let [sc (create-scripting-container 10080)]
-         (logutils/with-test-logging
-           (try
-             (.runScriptlet sc "response = c.get('/', {})")
-             (is (true? false) "Expected HTTP connection to HTTPS port to fail")
-             (catch EvalFailedException e
-               (is (instance? HttpClientException (.. e getCause)))
-               (is (instance? ConnectionClosedException (.. e getCause getCause)))))))))
+      (let [sc (create-scripting-container 10080)]
+        (logutils/with-test-logging
+          (try
+            (.runScriptlet sc (raise-caught-http-error "c.get('/', {})"))
+            (is false "Expected HTTP connection to HTTPS port to fail")
+           (catch EvalFailedException e
+             (is (instance? ConnectionClosedException (.getCause e)))))))))
```

Comment:
I actually kinda like it better this way anyway because it allows us to indirectly exercise the exception handling in Ruby, which is more interesting from a production code perspective than the Clojure code path.